source/common/network/connection_impl.cc

```diff
@@ -57,15 +58,24 @@ ConnectionImpl::ConnectionImpl(Event::DispatcherImpl& dispatcher, int fd,
                                Address::InstanceConstSharedPtr local_address,
                                Address::InstanceConstSharedPtr bind_to_address,
                                bool using_original_dst, bool connected)
+    : ConnectionImpl(dispatcher, fd, remote_address, local_address, bind_to_address,
+                     TransportSocketPtr{new RawBufferSocket}, using_original_dst, connected) {}
+
+ConnectionImpl::ConnectionImpl(Event::DispatcherImpl& dispatcher, int fd,
+                               Address::InstanceConstSharedPtr remote_address,
+                               Address::InstanceConstSharedPtr local_address,
+                               Address::InstanceConstSharedPtr bind_to_address,
+                               TransportSocketPtr transport_socket, bool using_original_dst,
```

Comment:
How much churn would it be to do away with the old constructor entirely?

Reply:
Not that much, but I prefer to land that in next PR, to keep this refactoring PRs small. Added a  TODO.